server/channels/app/channel_integration.go

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"net/http"
+	"os"
+	"sync"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/shared/request"
+)
+
+var (
+	integrationAdminUsername string
+	integrationAdminOnce     sync.Once
+)
+
+// getIntegrationAdminUsername returns the integration admin username from environment variable
+// using sync.Once to ensure it's only read once for performance.
+func getIntegrationAdminUsername() string {
+	integrationAdminOnce.Do(func() {
+		integrationAdminUsername = os.Getenv("INTEGRATION_ADMIN_USERNAME")
+	})
+	return integrationAdminUsername
+}
+
+// IsOfficialChannel checks if a channel is official by comparing creator with integration admin user.
+func (a *App) IsOfficialChannel(c request.CTX, channel *model.Channel) (bool, *model.AppError) {
+	if channel == nil {
+		return false, model.NewAppError("IsOfficialChannel", "app.channel.invalid", nil, "channel is nil", http.StatusBadRequest)
+	}
+
+	if channel.CreatorId == "" {
+		return false, nil
+	}
+
+	// Get cached integration admin username
+	adminUsername := getIntegrationAdminUsername()
+	if adminUsername == "" {
+		return false, model.NewAppError("IsOfficialChannel", "app.channel.config_missing", nil, "INTEGRATION_ADMIN_USERNAME not configured", http.StatusInternalServerError)
+	}
+
+	creatorUser, err := a.GetUser(channel.CreatorId)
+	if err != nil {
+		if err.StatusCode == http.StatusNotFound {
```

Comment:
Which implementation did you reference for this?

Reply:
```suggestion
		if err.Id == MissingAccountError {
```
It might be a good idea to use this standard pattern from mattermost.